data_steward/retraction/retract_utils.py

```diff
@@ -118,14 +110,6 @@ def get_datasets_list(client, dataset_ids_list):
             f"Datasets specified and existing in project {client.project}: {dataset_ids}"
         )
 
-    # consider datasets containing PPI/EHR data, excluding sandbox/staging datasets
-    dataset_ids = [
-        dataset_id for dataset_id in dataset_ids
-        if get_dataset_type(dataset_id) != OTHER and
-        not is_sandbox_dataset(dataset_id)
-    ]
```

Comment:
Thanks, @lrwb-aou. This is something I need your advice on.
I looked through the old sandbox datasets. But like you mentioned, I could not guarantee that every old sandbox dataset has reliable mapping/extension tables. Currently, with my code, the sandbox datasets are skipped after all if the retraction is `ONLY_EHR` here: https://github.com/all-of-us/curation/blob/hm/2904_2/data_steward/retraction/retract_data_bq.py#L399

The problems here are (1) some sandbox datasets have mapping/extension tables but some don't, and (2) some sandbox datasets have mapping/extension tables but their naming conventions are irregular. And I want to avoid the script being too complex to handle both (1) and (2). Do you have any suggestions how we should move forward?

Reply:
I would leave this exclusion in place.  Then we will know that the datasets we are looking to drop data from are more likely to contain conventionally named tables.

Staging datasets are intended to be dropped, so it is safe to leave this exclusion in place for them.

Sandbox datasets may deserve to have their own retraction rules that we develop later.  It can be an enhancement to this script.  For now, if a record has been sandboxed, it will not make it into a CDR.  I think we need to take the time to determine the best approach to the sandbox datasets.  It feels like that will be enough of a lift to deserve its own ticket.

Reply:
@hiro-mishima I just want to get your PR into the best possible place so it can be merged before you go on holiday.

Reply:
I put back the exclusion and updated the tests accordingly.